src/components/booklist/booklist.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import Book from '../book/book';
+import { connect } from 'react-redux';
+import './booklist.styles.scss'
+import { setLibrary } from '../../redux/actions/library';
+
+class BookList extends React.Component {
+
+	componentWillMount() {
+		fetch('http://localhost:3000/api/v1/books', {
+	      method: 'GET',
+	      headers: {'AUTHORIZATION': `Bearer ${sessionStorage.jwt}`}
+	    })
+		.then(res => res.json())
+		.then(data => this.props.setLibrary(data))
+		.catch(err => {console.log(err)});
+	}
+
+	render () {
+		const {books} = this.props;
+		return (
+			<div className='booklist'>
+				{ books.map(({id, title, category, read}) => (
+					<Book key={id} id={id} title={title} category={category} read={read} />
+				))
+				}
+			</div>
+	)}
+}
+
+const mapStateToProps = ({ library: {books}, filter: {filter} }) => ({
```

Comment:
You mentioned it's getting an error if we refactor it to the version I suggested in my previous review. Did we even investigate what caused such error? 

Also why do we need to force deconstruct? Why not simply:

```javascript
const mapStateToProps = state => ({
 books: state.books.filter(book => state.filter.length === 0 ? true : book.category === state.filter),
  filter: state.filter
})
```
?

Again my message here is let us not overcomplicate our lives :)